release.sh

```diff
@@ -0,0 +1,171 @@
+fi
+
+# Read current version information from version file
+defaultMajor="$(sed -n ${majorLine}p ${versionFile})"
+defaultMinor="$(sed -n ${minorLine}p ${versionFile})"
+defaultFix="$(sed -n ${fixLine}p ${versionFile})"
+defaultDS="$(sed -n ${DSLine}p ${versionFile})"
+defaultCommit="$(sed -n ${commitLine}p ${versionFile})"
+currentVer=${defaultMajor}.${defaultMinor}.${defaultFix}.${defaultDS}.${defaultCommit}
+
+# Ask user to input new version information
+echo -e "Current software version:  ${currentVer}"
+read -p "Please enter major version [${defaultMajor}]: " major
+major=${major:-${defaultMajor}}
+read -p "Please enter minor version [${defaultMinor}]: " minor
+minor=${minor:-${defaultMinor}}
+read -p "Please enter fix version [${defaultFix}]: " fix
+fix=${fix:-${defaultFix}}
+read -p "Please enter expected DS epoch [${defaultDS}]: " DS
+DS=${DS:-${defaultDS}}
+commit="$(git describe --always)"
+
+# Write new version information into version file
+echo -e "Writing new software version into ${versionFile}..."
+sed -i "${majorLine}s/.*/${major}/" ${versionFile}
+sed -i "${minorLine}s/.*/${minor}/" ${versionFile}
+sed -i "${fixLine}s/.*/${fix}/" ${versionFile}
+sed -i "${DSLine}s/.*/${DS}/" ${versionFile}
+sed -i "${commitLine}s/.*/${commit}/" ${versionFile}
+newVer=${major}.${minor}.${fix}.${DS}.${commit}
+export ZIL_VER=${newVer}
+export ZIL_PACK_NAME=${packageName}
+echo -e "New software version: ${newVer} is written into ${versionFile} successfully.\n"
+
+# Use cpack to making deb file
+echo -e "Make deb package..."
+rm -rf build; rm -rf build_lookup
+./build.sh; cd build; make package; cd -
+./build_lookup.sh; cd build_lookup; make package; cd -
+cp ${versionFile} build/; cp ${versionFile} build_lookup/
+cd build; debFile="$(ls *.deb)"; cd -
+echo -e "Deb packages are generated successfully.\n"
+
+# Make SHA-256 & multi-signature
+echo -e "Making SHA-256 & multi-signature..."
+privKeyFile="$(realpath $1)"
+pubKeyFile="$(realpath $2)"
+cd build
+sha="$(md5sum ${debFile}|cut -d ' ' -f1)"
+sed -i "${shaLine}s/.*/${sha}/" ${versionFile}
+signature="$(./bin/sign ${sha} ${privKeyFile} ${pubKeyFile})"
+sed -i "${sigLine}s/.*/${signature}/" ${versionFile}
+cd -
+cd build_lookup
+sha="$(md5sum ${debFile}|cut -d ' ' -f1)"
+sed -i "${shaLine}s/.*/${sha}/" ${versionFile}
+signature="$(./bin/sign ${sha} ${privKeyFile} ${pubKeyFile})"
+sed -i "${sigLine}s/.*/${signature}/" ${versionFile}
+cd -
+echo -e "SHA-256 & multi-signature are written into ${versionFile} successfully.\n"
+
+# Upload package onto GitHub
+echo -e "Creating new release and uploading package onto GitHub..."
+fullCommit="$(git rev-parse HEAD)"
+curl -v -s \
+  -H "Authorization: token ${GitHubToken}" \
+  -H "Content-Type:application/json" "https://api.github.com/repos/Zilliqa/${repoName}/releases" \
+  -d '{
+  "tag_name": "'"${newVer}"'", 
+  "target_commitish": "'"${fullCommit}"'",
+  "name": "'"${releaseTitle}"'",
+  "body": "'"${releaseDescription}"'",
+  "draft": false,
```

Comment:
Perhaps we could utilize this option to have a more robust release workflow:

1. create a draft release with this option `draft` set to `true`
2. upload the assets
3. verify all the assets are successfully uploaded. If there's any error, go to step 2
4. publish the draft release